twine/commands/upload.py

```diff
@@ -48,6 +53,39 @@ def skip_upload(response, skip_existing, package):
             (response.status_code == 403 and msg_403 in response.text)))
 
 
+def check_status_code(response, verbose):
+    """
+    Additional safety net to catch response code 410 in case the
+    UploadToDeprecatedPyPIDetected exception breaks.
+    Also includes a check for response code 405 and prints helpful error
+    message guiding users to the right repository endpoints.
+    """
+    if (response.status_code == 410 and
+            response.url.startswith(("https://pypi.python.org",
+                                     "https://testpypi.python.org"))):
+        print("It appears you're uploading to pypi.python.org (or "
+              "testpypi.python.org). You've received a 410 error response. "
+              "Uploading to those sites is deprecated. The new sites are "
+              "pypi.org and test.pypi.org. Try using "
+              "https://upload.pypi.org/legacy/ "
```

Comment:
I think this and the line below are other good spots to use the `*_REPOSITORY` constants.

Reply:
Fixed in `ea8d010`.